### PR TITLE
fix: migrate well known to openapi

### PR DESCRIPTION
**Pull request by markusahlstrand**
Moved the well known endpoint. Some issue with strict typing of the return values..

### CODE REVIEW

File: src/routes/oauth2/well-known.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  // --------------------------------
+  .openapi(
+    createRoute({
+      tags: ["jwks"],
+      method: "get",
+      path: "/jwks.json",
+      request: {},
+      responses: {
+        200: {
+          // TODO: not sure why this types doesn't match
```
Comment (markusahlstrand) [inline src/routes/oauth2/well-known.ts:18]: I think this is some other dependency that we have that causes this.. Works fine in the api-proxy with the same code

File: test/integration/jwks.spec.ts [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 import { tsoaApp } from "../../src/app";
+import { wellKnown } from "../../src/routes/oauth2/well-known";
 import { testClient } from "hono/testing";
+import { z } from "zod";
 
 describe("jwks", () => {
   it("should return a list with the test certificate", async () => {
     const env = await getEnv();
-    const client = testClient(tsoaApp, env);
+    const client = testClient(wellKnown, env);
```
Comment (markusahlstrand) [inline test/integration/jwks.spec.ts:16]: This is slightly cheating.. I'm pulling the route directly which made it easier to get the types working. I think we long term should use the main app instead.

File: src/routes/oauth2/well-known.ts (hunk 1 above)
Comment (danielgent) [reply src/routes/oauth2/well-known.ts:18]: The comments should say `this type` or `these types`, but I won't block merging a PR for that :smile:

Comment (danielgent) [reply src/routes/oauth2/well-known.ts:18]: I'll be back in england next month so I better get used to totally incoherent written English again :smiling_face_with_tear:

File: test/integration/jwks.spec.ts (hunk 2 above)
Comment (danielgent) [reply test/integration/jwks.spec.ts:16]: What's a long term solution?  See https://github.com/sesamyab/auth/blob/main/test/integration/flows/password.spec.ts#L768

I did

```
 const anyClient = client as any;

      // NOTE - I'm not testing the GET that loads the webform here... we don't have a browser to interact with here
      const resetPassword = await anyClient.u["reset-password"].$post({
```


The route is there of course just not the types... Do we need to migrate everything first? 

Main thing is the tests are still working

Comment (danielgent) [reply test/integration/jwks.spec.ts:16]: Can we turn `export const tsoaApp = RegisterRoutes(app as unknown as Hono);` into some kind of

```
export const tsoaAppTesting = {...tsoaApp, loginApp}
```

:question:

Comment (danielgent) [reply test/integration/jwks.spec.ts:16]: We probably want to fix this *now* else we'll forget about these `any`s or specific usage of routes

Comment (markusahlstrand) [reply test/integration/jwks.spec.ts:16]: The question is if we should have one big app or if we should have multiple? Think that we should try to separate the management api from the rest at least..

Comment (danielgent) [reply test/integration/jwks.spec.ts:16]: But that should be done separately to migrating surely?

If we have to change the tests everytime we change a route then what are we doing here?  :smile: 

We're testing implementation details really.

We can just drop the types here by using `any` until we work something out (as mentioned above). I'd rather manually add types for now somehow as the passing tests are the real test! :smile: 

Your changes here could easily have broken things but as we're just testing the route now, we'll get :heavy_check_mark:

Comment (danielgent) [reply test/integration/jwks.spec.ts:16]: I had a go and remembered why I couldn't do this easily last time :smile: 

TBD but I don't want to block merging this :sunglasses:

**Pull request merged**